### PR TITLE
Web Api interface generation

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
**Description** (*)
This PR adds the generation of a Web API interface for a service.

**What was done:**

1. generation of a Web API interface for a service.
2. added possibility to choose one or more methods to be included in an interface
3. added possibility to choose interface name and change its description (default one is taken from the service)
4. all methods short description taken from the service methods (or default TODO, that asks to add short description)
5. implements added for a service
6. implements class imported in use block for a service
7. all included in interface methods annotations replaced with @inheritDoc annotations.
8. added test case to test simple service with primitive types
9. added test case to test simple service with Object types
10. testing for inserting @inheritDoc annotations
11. testing for an interface implemented by a service
12. testing for interface FQN added in an import (use) block

**Service to add Web Api interface:**
Screenshot 2021-05-03 at 13 45 47

**Open the context menu on it and choose Web API interface generation action:**
Screenshot 2021-05-03 at 13 47 31

**Choose methods to include in the interface, change default name and description:**
Screenshot 2021-05-03 at 13 48 17
Screenshot 2021-05-03 at 13 48 56

**Results:**
Screenshot 2021-05-03 at 13 49 25
Screenshot 2021-05-03 at 13 50 09
Screenshot 2021-05-03 at 13 51 37

**Questions or comments**
To get generated Doc Annotations beautiful, go to the `Preferences... > Editor > Code Style > PHP > PHPDoc > Blank lines around parameters` and activate this configuration.

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @bohdan-harniuk, thank you for your contribution. Could you please rename all the visible text occurrences of `Api` to `API`, since [API](https://en.wikipedia.org/wiki/API) is an acronym and should be correctly used. Below you may see couple of places, but would be good to change it on every place that is visible to the user.

Also, can we exclude these API generators for `Test/` or probably `*Test.php` files, in order to not overload the context menu? I don't see any use case, when we have to create an API for any kind of tests.

Thank you.

File: resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -260,6 +263,8 @@
         <internalFileTemplate name="Magento Delete Entity By Id Command"/>
         <internalFileTemplate name="Magento Entity Edit Action Controller Class"/>
         <internalFileTemplate name="Magento Entity Delete Controller Class"/>
+        <internalFileTemplate name="Magento Web Api XML"/>
+        <internalFileTemplate name="Web Api Interface"/>
```
[the remark this answers is not in the document]
Comment (bohdan-harniuk) [reply resources/META-INF/plugin.xml:267]: Done.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewInterfaceForServiceDialog.form [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            <constraints>
+              <grid row="7" column="1" row-span="1" col-span="1" vsize-policy="6" hsize-policy="1" anchor="0" fill="2" indent="0" use-parent-layout="false"/>
+            </constraints>
+          </vspacer>
+          <component id="7370" class="javax.swing.JLabel" binding="serviceClassFieldLabel">
+            <constraints>
+              <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+            </constraints>
+            <properties>
+              <text value="Api For Class"/>
```
[the remark this answers is not in the document]
Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/NewInterfaceForServiceDialog.form:83]: Done.

**Comment by bohdan-harniuk** [comment]
Thank you, @eduard13, for the great findings! I've fixed them.

Screenshot 2021-05-05 at 10 57 38

**Review by eduard13 — APPROVED** [review]
Thank you for your contribution 👍

**Pull request merged**